master/internal/core.go

```diff
@@ -1039,6 +1041,19 @@ func (m *Master) Run(ctx context.Context) error {
 	tasksGroup := m.echo.Group("/tasks")
 	tasksGroup.GET("", api.Route(m.getTasks))
 
+	if m.config.InternalConfig.LorePath != "" {
+		lorePrefix := "/lore"
+		loreGroup := m.echo.Group(lorePrefix)
+
+		loreTarget, err := url.Parse(m.config.InternalConfig.LorePath)
+		if err != nil {
+			return errors.Wrap(err, "failed to parse the given Lore path")
+		}
+
+		loreProxy := httputil.NewSingleHostReverseProxy(loreTarget)
```

Comment:
discuss: do we want to whitelist only a few set headers?

Reply:
I don't quite know the full ramifications security or otherwise of allowing all headers, but to the best of my understanding it is probably alright to pass them all for now? This proxy would only be internal and not exposed for the moment. I would think that for a more refined beta deployment this will be something we should be considering more carefully